### DISCUSSION TITLE
Tweedie + quadratic?

### DISCUSSION THREAD

**Discussion by gerverska** [JenniNiku/gllvm#182] [Ideas]
Hi folks,

I noticed that current versions of `gllvm` have the following `stop()` messages for Tweedie models:
```
The quadratic model is not implemented for tweedie family yet
```
Are there plans to enable quadratic models specifying a Tweedie distribution? If this isn't a feature priority at the moment, do any potential workarounds exist?

Thanks!

**Comment by BertvanderVeen** [comment]
Thanks for inquiring!

The maths for a quadratic response model is a bit more involved, and it does not always work out. To fit Tweedie class models, gllvm uses a combination of Laplace and Variational approximations, but the Laplace approximation does not work for a quadratic response model (also noted in van der Veen et al. 2021). So, unfortunately, it will not (any time soon, that is) be possible to fit since it requires a new/completely different fitting method not already implemented in the package.

It might be possible to find a suitable workaround though, what type of data are you trying to fit the model to?

**Reply by gerverska** [reply]
Thanks, Bert, good to know. I don't have a pressing need at the moment, but I was curious about the extent of the difficulty and where this functionality ranked for team `gllvm`.

I have microbial load indices (many 0s, and then positive continuous responses) for less than 100 taxa that I'm trying to model. When looking at the residuals, there are some taxa that seem like they might benefit from the addition of a quadratic term, but I'm still classifying the extent of the issue.

**Reply by BertvanderVeen** [reply]
Woops, I was wrong. I said this last year, but I  implemented the Tweedie distribution with VA mid last year, so that it is available for the quadratic model.
